integration-metrics/src/main/java/org/symphonyoss/integration/metrics/IntegrationMetricsController.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2016-2017 Symphony Integrations - Symphony LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.symphonyoss.integration.metrics;
+
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Controller class to delegate the metric objects creation to the specific controller.
+ * Created by rsanchez on 12/12/16.
+ */
+@Component
+public class IntegrationMetricsController {
+
+  /**
+   * Holds the list of controller to be initialized
+   */
+  private List<IntegrationController> controllers = new ArrayList<>();
```

Comment:
You could also autowire this collection.  Spring will initialize the collection with all beans that implement IntegrationController interface.  Then controllers wouldn't need to self register.  Downside to autowiring is that new controllers cannot be registered after container is initialized.